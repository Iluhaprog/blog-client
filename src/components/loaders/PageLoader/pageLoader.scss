@import '../../../vars.scss';

@keyframes anim {
    0% {
        transform: scaleY(1);
    }
    50% {
        transform: scaleY(3);
    }
    100% {
        transform: scaleY(1);
    }
}

.loader {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-wrap: wrap;
    height: 100vh;
    background-color: #fff;
}

.page-loader {
    max-width: 300px;
    min-height: 300px;
    width: 100%;
    border: $border-1;
    border-radius: 50%;
    box-shadow: $shadow;
    display: flex;
    flex-direction: column;
    justify-content: center;

    &__title {
        font-weight: 300;
        font-size: 40px;
        text-align: center;
        width: 100%;
        margin: 0;
    }

    &__subtitle {
        font-weight: 300;
        text-align: center;
        margin-bottom: 40px;
    }

    &__animation {
        max-width: 200px;
        display: flex;
        justify-content: center;
        margin: 0 auto;

        span {
            display: block;
            margin: 10px;
            width: 20px;
            height: 20px;
            background-color: $nav-bg;
            opacity: .9;
            border-radius: 2px;
            animation-name: anim;
            animation-duration: 1s;
            animation-iteration-count: infinite;
            animation-timing-function: ease-in-out;

            &:nth-child(1) {

            }

            &:nth-child(2) {
                animation-delay: .250s;
            }

            &:nth-child(3) {
                animation-delay: .5s;
            }
        }
    }
}