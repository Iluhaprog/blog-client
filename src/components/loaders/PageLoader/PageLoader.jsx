import React from 'react';
import PropTypes from 'prop-types';
import { Loader } from '../Loader';
import './pageLoader.scss';

function PageLoader(props) {
    return (
        <Loader visible={props.visible} className='loader__big'>
            <div className='page-loader'>
                <h2 className='page-loader__title'>Hello!</h2>
                <h3 className='page-loader__subtitle'>
                    Welcome to my blog!
                </h3>
                <div className='page-loader__animation'>
                    <span></span>
                    <span></span>
                    <span></span>
                </div>
            </div>
        </Loader>
    );
}

PageLoader.propTypes = {
    visible: PropTypes.bool,
};

export default PageLoader;