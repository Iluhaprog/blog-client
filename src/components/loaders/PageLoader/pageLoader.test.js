import React from 'react';
import { render } from '@testing-library/react';
import PageLoader from './PageLoader';

test('Should render PageLoader component', () => {
    const { asFragment } = render(<PageLoader visiple={true}/>);
    expect(asFragment()).toMatchSnapshot();
});