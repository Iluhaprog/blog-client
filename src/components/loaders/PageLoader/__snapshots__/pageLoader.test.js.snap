// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render PageLoader component 1`] = `
<DocumentFragment>
  <div
    class="loader "
  >
    <div
      class="page-loader"
    >
      <h2
        class="page-loader__title"
      >
        Hello!
      </h2>
      <h3
        class="page-loader__subtitle"
      >
        Welcome to my blog!
      </h3>
      <div
        class="page-loader__animation"
      >
        <span />
        <span />
        <span />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
